dist-primitives/src/utils/pack.rs

```diff
@@ -1,21 +1,34 @@
 use ark_ff::FftField;
+use ark_poly::domain::DomainCoeff;
+use ark_std::cfg_chunks;
 use rand::thread_rng;
 use secret_sharing::pss::PackedSharingParams;
 
+pub fn best_unpack<F: FftField, T: DomainCoeff<F>>(
+    shares: &[T],
+    parties: &[u32],
+    pp: &PackedSharingParams<F>,
+) -> Vec<T> {
+    debug_assert_eq!(shares.len(), parties.len());
+    if shares.len() == pp.n {
+        pp.unpack2(shares.to_vec())
+    } else {
+        pp.lagrange_unpack(shares, parties)
+    }
+}
```

Comment:
Can we have this free function as a method on `PackedSharingParams`? it would act the same actually and would make it easier to use (passing 2 values instead of 3).

Reply:
Yes so there's two lines of thought I had here:
1. If we make best_unpack a part of the secret-sharing crate, then we force the `best_unpack` method on anyone using the secret-sharing crate. For eg. at small values of `n` largrange_unpack always runs faster than unpack2, even though it's asymptotically more expensive.
2. If we don't make it a part of the secret-sharing crate as is done currently, we offer flexibility to the developer but this requires passing the 3rd parameter as you observed.

Happy to change to the 1st approach if you feel it's better.

Reply:
I see, but why not both? As in, we offer the high-level `best_unpack` which picks the best function for you, and if you really know what you are doing you decide yourself and use the other low-level functions.

Reply:
Changed `best_unpack` to `pp.unpack_missing_shares()` -- name changed to avoid confusion that this would offer the best performance.